extensions/binding/org.eclipse.smarthome.binding.astro/src/main/java/org/eclipse/smarthome/binding/astro/internal/job/Job.java

```diff
@@ -0,0 +1,218 @@
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.eclipse.smarthome.binding.astro.internal.job;
+
+import static java.util.Objects.isNull;
+import static org.apache.commons.lang.time.DateFormatUtils.ISO_DATETIME_FORMAT;
+import static org.eclipse.smarthome.binding.astro.AstroBindingConstants.*;
+import static org.eclipse.smarthome.binding.astro.internal.util.CronHelper.createCronFromCalendar;
+import static org.eclipse.smarthome.binding.astro.internal.util.DateTimeUtils.*;
+
+import java.lang.invoke.MethodHandles;
+import java.util.Calendar;
+
+import org.eclipse.smarthome.binding.astro.handler.AstroThingHandler;
+import org.eclipse.smarthome.binding.astro.internal.config.AstroChannelConfig;
+import org.eclipse.smarthome.binding.astro.internal.model.Planet;
+import org.eclipse.smarthome.binding.astro.internal.model.Range;
+import org.eclipse.smarthome.binding.astro.internal.model.SunPhaseName;
+import org.eclipse.smarthome.core.scheduler.CronExpression;
+import org.eclipse.smarthome.core.scheduler.Expression;
+import org.eclipse.smarthome.core.scheduler.ExpressionThreadPoolManager.ExpressionThreadPoolExecutor;
+import org.eclipse.smarthome.core.thing.Channel;
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.binding.ThingHandler;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The interface to be implemented by classes which represent a 'job' to be
+ * performed.
+ *
+ * @author Amit Kumar Mondal - New Simplified API, Implementation
+ *         compliant with ESH Scheduler
+ */
+public interface Job extends Runnable {
+
+    /** Logger Instance */
+    public final Logger logger = LoggerFactory.getLogger(MethodHandles.lookup().lookupClass());
+
+    /**
+     * Schedules the provided {@link Job} instance
+     *
+     * @param thingUID
+     *            the UID of the {@link Thing} instance
+     * @param astroHandler
+     *            the {@link AstroThingHandler} instance
+     * @param job
+     *            the {@link Job} instance to schedule
+     * @param eventAt
+     *            the {@link Calendar} instance denoting scheduled instant
+     */
+    public static void schedule(String thingUID, AstroThingHandler astroHandler, Job job,
+            Calendar eventAt) {
+        try {
+            Calendar today = Calendar.getInstance();
+            if (isSameDay(eventAt, today) && isTimeGreaterEquals(eventAt, today)) {
+                ExpressionThreadPoolExecutor executor = astroHandler.getScheduler();
+                if (executor != null) {
+                    Expression cron = new CronExpression(createCronFromCalendar(eventAt));
+                    executor.schedule(job, cron);
+                    astroHandler.addJobToQueue(job);
```

Comment:
Again: If this job is not added to your queue, then do not schedule it in the scheduler.